src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -408,7 +408,7 @@ The .NET Foundation licenses this file to you under the MIT license.
     <ItemGroup>
       <ExecProjects Include="$(MSBuildThisFileDirectory)ExecWrapper.proj">
         <AdditionalProperties>
-          ExecCommand=$(WasmCompilerPath) &quot;%(LlvmObjects.Identity)&quot; -o &quot;%(LlvmObjects.NativeObject)&quot; $(CompileWasmArgs)
+          ExecCommand=$(WasmCompilerPath) &quot;$(ProjectDir)/%(LlvmObjects.Identity)&quot; -o &quot;$(ProjectDir)/%(LlvmObjects.NativeObject)&quot; $(CompileWasmArgs)
```

Comment:
These should should already be full paths, derived from `ManagedBinary.IlcOutputFile`. Rather mysterious they were not in your case.

Reply:
I see the CI is failing in line with your comment.  However `IlcOutputFile` is based on  `NativeIntermediateOutputPath`

```xml
        <IlcOutputFile>$(NativeIntermediateOutputPath)\%(Filename)$(IlcOutputFileExt)</IlcOutputFile>
```
```xml
    <NativeIntermediateOutputPath Condition="'$(NativeIntermediateOutputPath)' == ''">$(IntermediateOutputPath)native\</NativeIntermediateOutputPath>
```
```xml
    <IntermediateOutputPath Condition=" $(IntermediateOutputPath) == '' and '$(PlatformName)' == 'AnyCPU' ">$(BaseIntermediateOutputPath)$(Configuration)\</IntermediateOutputPath>
    <IntermediateOutputPath Condition=" $(IntermediateOutputPath) == '' and '$(PlatformName)' != 'AnyCPU' ">$(BaseIntermediateOutputPath)$(PlatformName)\$(Configuration)\</IntermediateOutputPath>
    <IntermediateOutputPath Condition="!HasTrailingSlash('$(IntermediateOutputPath)')">$(IntermediateOutputPath)\</IntermediateOutputPath>
```
```xml
    <BaseIntermediateOutputPath Condition="'$(BaseIntermediateOutputPath)'=='' ">obj\</BaseIntermediateOutputPath>
```
Things would be different if `UseArtifactsOutput` was `true` but it is not set.

When the tests run they do get from `microsoft.dotnet.arcade.sdk\8.0.0-beta.23320.3\tools\RepoLayout.props`

```xml
    <ArtifactsDir Condition="'$(ArtifactsDir)' == ''">$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts'))</ArtifactsDir>
```
And that changes `BaseIntermediateOutputPath ` to be absolute

The difference seems to be the .net 8 sdk versions.  Locally I have  8.0.100-preview.6.23330.14 which in its `SDK.props` does not have a 
```xml
  <Import Project="..\tools\Settings.props"
```
which is the root of setting `ArtifactsDir ` in `\8.0.0-beta.23320.3` used in CI

Reply:
I see, thank you. The `ExecWrapper.proj` runs in its own context.

I think it would make sense to set the `WorkingDirectory` on the `Exec` task explicitly by passing it down through an additional property:
```xml
<ExecProjects Include="$(MSBuildThisFileDirectory)ExecWrapper.proj">
  <AdditionalProperties>
    ExecCommand=$(WasmCompilerPath) &quot;%(LlvmObjects.Identity)&quot; -o &quot;%(LlvmObjects.NativeObject)&quot; $(CompileWasmArgs);
    ExecWorkingDirectory=$(MSBuildProjectDirectory)
  </AdditionalProperties>
</ExecProjects>
```